src/acceptance-test/java/net/consensys/orion/acceptance/dsl/OrionInstance.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package net.consensys.orion.acceptance.dsl;
+
+import net.consensys.cava.crypto.sodium.Box;
+import net.consensys.cava.crypto.sodium.Box.PublicKey;
+import net.consensys.cava.io.Base64;
+import net.consensys.orion.http.handler.receive.ReceiveRequest;
+import net.consensys.orion.http.server.HttpContentType;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import io.vertx.core.json.JsonObject;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+/**
+ * Responsible for providing access to a running Orion instance via its HTTP interface such that payloads can be
+ * submitted and extracted.
+ *
+ * It also encapsulates all aspects of the Orion Instance - such as its datapath and maintained keys.
+ */
+public class OrionInstance {
+
+  private static final String configFileName = "config.toml";
+  private static final String networkInterface = "127.0.0.1";
+
+  private static final String urlPattern = "http://" + networkInterface + ":%d";
+
+  private final String nodeName;
+  private final List<KeyDefinition> keys;
+  private final Path workPath;
+  private final String libSodiumPath;
+  private final List<String> bootnodeClientUrl;
+  private OkHttpClient httpClient;
+
+  private OrionProcessRunner runner = null;
+
+  public OrionInstance(
+      final String nodeName,
+      final List<KeyDefinition> keys,
+      final Path workPath,
+      final String libSodiumPath,
+      final List<String> bootnodeClientUrl) {
+    this.nodeName = nodeName;
+    this.keys = keys;
+    this.workPath = workPath;
+    this.libSodiumPath = libSodiumPath;
+    this.bootnodeClientUrl = bootnodeClientUrl;
+  }
+
+  public void start() throws IOException {
+    generateConfigFile();
+    runner = new OrionProcessRunner(configFileName, workPath);
+    runner.start(nodeName);
+    httpClient = new OkHttpClient();
+
+  }
+
+  public String sendData(final byte[] data, final Box.PublicKey sender, final Collection<PublicKey> recipients)
```

Comment:
Method overloading might be an idea, as in at least one of the tests you're sending to all the connections the Orion has, using a method that assumes sending to all connections would reduce the noise in the tests.
`public String sendData(final byte[] data, final Box.PublicKey sender)`

Reply:
need to really make use of json serialise/deserialise of the Packet types - then break them out to an api module, THEN we can get Pantheon and AT to use the same things